File: src/main/ejemplo-angular/src/app/permiso-atributos/shared/permiso-atributo.ts
/**
 *
 * @author IstmoSoft
 */
export class PermisoAtributo {

// ids inicio
      id = null ;
// ids fin

// atributos inicio
      etiqueta = null ;
      validacion = null ;
      permiso = null ;
      tabla = null ;
      atributo = null ;
      mensaje = null ;
      status = null ;
      usuarioUltMov = null ;
      usuario = null ;
      perfiles = null ;
// atributos fin
   constructor (source = null) {
      if (source!== null)
         this.from(source);
   }
  from(source) {
      if( source.id)
           this.id = source.id;
      if( source.etiqueta)
           this.etiqueta = source.etiqueta;
      if( source.validacion)
           this.validacion = source.validacion;
      if( source.permiso)
           this.permiso = source.permiso;
      if( source.tabla)
           this.tabla = source.tabla;
      if( source.atributo)
           this.atributo = source.atributo;
      if( source.mensaje)
           this.mensaje = source.mensaje;
      if( source.status)
           this.status = source.status;
      if( source.usuarioUltMov)
           this.usuarioUltMov = source.usuarioUltMov;
      if( source.usuario)
           this.usuario = source.usuario;
      if( source.perfiles)
           this.perfiles = source.perfiles;

  }
  isOk() {
      if( this.id)
           return true;
      if( this.etiqueta)
           return true;
      if( this.validacion)
           return true;
      if( this.permiso)
           return true;
      if( this.tabla)
           return true;
      if( this.atributo)
           return true;
      if( this.mensaje)
           return true;
      if( this.status)
           return true;
      if( this.usuarioUltMov)
           return true;
      if( this.usuario)
           return true;
      if( this.perfiles && this.perfiles.length >0   )
           return true;

    return false;
  }
  toDto() {
     if ( this.id !=null && typeof this.id === 'string' && this.id.replace(/^\s+|\s+$/g, '')=="" )
          this.id = null;
     if ( this.etiqueta !=null && typeof this.etiqueta === 'string' && this.etiqueta.replace(/^\s+|\s+$/g, '')=="" )
          this.etiqueta = null;
     if ( this.validacion !=null && typeof this.validacion === 'string' && this.validacion.replace(/^\s+|\s+$/g, '')=="" )
          this.validacion = null;
     if ( this.permiso !=null && typeof this.permiso === 'string' && this.permiso.replace(/^\s+|\s+$/g, '')=="" )
          this.permiso = null;
     if ( this.tabla !=null && typeof this.tabla === 'string' && this.tabla.replace(/^\s+|\s+$/g, '')=="" )
          this.tabla = null;
     if ( this.atributo !=null && typeof this.atributo === 'string' && this.atributo.replace(/^\s+|\s+$/g, '')=="" )
          this.atributo = null;
     if ( this.mensaje !=null && typeof this.mensaje === 'string' && this.mensaje.replace(/^\s+|\s+$/g, '')=="" )
          this.mensaje = null;
     if ( this.status !=null && typeof this.status === 'string' && this.status.replace(/^\s+|\s+$/g, '')=="" )
          this.status = null;
     if ( this.usuarioUltMov !=null && typeof this.usuarioUltMov === 'string' && this.usuarioUltMov.replace(/^\s+|\s+$/g, '')=="" )
          this.usuarioUltMov = null;
     if ( this.usuario !=null && typeof this.usuario === 'string' && this.usuario.replace(/^\s+|\s+$/g, '')=="" )
          this.usuario = null;
     if ( this.perfiles !=null && typeof this.perfiles === 'string' && this.perfiles.replace(/^\s+|\s+$/g, '')=="" )
          this.perfiles = null;

  return this;
  }

}

File: src/main/java/mx/atto/ejemplo/entity/PermisoAtributo.java
package mx.atto.ejemplo.entity;

import java.io.Serializable;
import java.util.Date;
import java.util.HashSet;
import java.util.Set;
import java.util.Iterator;
import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.ForeignKey;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.JoinTable;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
//import mx.edu.wendulinea.control.entity.core.Usuario;

/**
 *
 * @author IstmoSoft
 */
@Entity
@Table(name = "permiso_atributo")
public class PermisoAtributo {

// ids inicio
private Integer id = null ;
// ids fin

// atributos inicio
private String etiqueta = null ;
private String validacion = null ;
private java.lang.Integer permiso = null ;
private String tabla = null ;
private String atributo = null ;
private String mensaje = null ;
private java.lang.Integer status = null ;
private Usuario usuarioUltMov = null ;
private Usuario usuario = null ;
private Set<Perfil> perfiles = new HashSet<Perfil>(0);
// atributos fin

// constructores inicio
     public PermisoAtributo() {
     }
     public PermisoAtributo( Integer id,  String etiqueta,  String validacion,  java.lang.Integer permiso,  String tabla,  String atributo,  String mensaje,  java.lang.Integer status,  Set<Perfil> perfiles,  Usuario usuarioUltMov,  Usuario usuario) {

          this.id=id;
          this.etiqueta=etiqueta;
          this.validacion=validacion;
          this.permiso=permiso;
          this.tabla=tabla;
          this.atributo=atributo;
          this.mensaje=mensaje;
          this.status=status;
          this.perfiles=perfiles;
          this.usuarioUltMov=usuarioUltMov;
          this.usuario=usuario;
     }
// constructores fin

// geters inicio
    @Id
    @GeneratedValue
    public Integer getId() {
        return id;
    }
 @Column(name = "etiqueta")
    public String getEtiqueta() {
        return etiqueta;
    }
 @Column(name = "validacion")
    public String getValidacion() {
        return validacion;
    }
 @Column(name = "permiso")
    public java.lang.Integer getPermiso() {
        return permiso;
    }
 @Column(name = "tabla")
    public String getTabla() {
        return tabla;
    }
 @Column(name = "atributo")
    public String getAtributo() {
        return atributo;
    }
 @Column(name = "mensaje")
    public String getMensaje() {
        return mensaje;
    }
 @Column(name = "status")
    public java.lang.Integer getStatus() {
        return status;
    }
    @ManyToOne
    @JoinColumn(name = "usuarioultmov", foreignKey = @ForeignKey(name = "fk_usuario"))
    public Usuario getUsuarioUltMov() {
        return usuarioUltMov;
    }
    @ManyToOne
    @JoinColumn(name = "usuario", foreignKey = @ForeignKey(name = "fk_usuario"))
    public Usuario getUsuario() {
        return usuario;
    }
     @ManyToMany(
            cascade = {CascadeType.PERSIST, CascadeType.MERGE},
            targetEntity = Perfil.class
    )
    @JoinTable(
            name = "permisoatributo_perfil",
            foreignKey = @ForeignKey(name = "pk"),
            joinColumns = @JoinColumn(name = "permisoatributo_id" , foreignKey = @ForeignKey(name = "fk_permisoatributo") ),
            inverseJoinColumns = @JoinColumn(name = "perfil_id"  , foreignKey = @ForeignKey(name = "fk_perfil"))
    )
    public Set<Perfil> getPerfiles() {
        return perfiles;
    }

    public void setId(Integer id) {
        this.id = id;
    }
    public void setEtiqueta(String etiqueta) {
        this.etiqueta = etiqueta;
    }
    public void setValidacion(String validacion) {
        this.validacion = validacion;
    }
    public void setPermiso(java.lang.Integer permiso) {
        this.permiso = permiso;
    }
    public void setTabla(String tabla) {
        this.tabla = tabla;
    }
    public void setAtributo(String atributo) {
        this.atributo = atributo;
    }
    public void setMensaje(String mensaje) {
        this.mensaje = mensaje;
    }
    public void setStatus(java.lang.Integer status) {
        this.status = status;
    }
    public void setPerfiles(Set<Perfil> perfiles) {
        this.perfiles = perfiles;
    }
    public void setUsuarioUltMov(Usuario usuarioUltMov) {
        this.usuarioUltMov = usuarioUltMov;
    }
    public void setUsuario(Usuario usuario) {
        this.usuario = usuario;
    }
// seters fin
}
